### ISSUE TITLE
v2.2.6 Error with Walk of Life: Immersive Professions

### ISSUE THREAD

**Issue by lightsourced** [Annosz/UIInfoSuite2#231]
https://smapi.io/log/78b7d5d3d8e24949b72feae67d8062c8

Here's my log! :) -- Also reported to Walk of Life: Immersive Professions (https://www.nexusmods.com/stardewvalley/mods/8111)

**Comment by argon932** [comment]
Same issue.

[Walk Of Life - Immersive Professions] [Harmonizer]: Failed to apply ExperieneBarDrawExperienceBarPatch.
Harmony returned System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.Exception: Couldn't find type named UIInfoSuite.UIElements.ExperienceBar.
   at DaLion.Common.Extensions.Reflection.StringExtensions.ToType(String name) in C:\dev\sdv-mods\ImmersiveValley\Common\Extensions\Reflection\StringExtensions.cs:line 15
   at DaLion.Stardew.Professions.Framework.Patches.Integrations.UiInfoSuite.ExperieneBarDrawExperienceBarPatch..ctor() in C:\dev\sdv-mods\ImmersiveValley\ImmersiveProfessions\Framework\Patches\Integrations\UiInfoSuite\ExperienceBarDrawExperienceBarPatch.cs:line 21
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at DaLion.Common.Harmony.Harmonizer.ApplyAll() in C:\dev\sdv-mods\ImmersiveValley\Common\Harmony\Harmonizer.cs:line 67
[Walk Of Life - Immersive Professions] [Harmonizer]: Failed to apply ExperienceBarGetExperienceRequiredToLevelPatch.
Harmony returned System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.Exception: Couldn't find type named UIInfoSuite.UIElements.ExperienceBar.
   at DaLion.Common.Extensions.Reflection.StringExtensions.ToType(String name) in C:\dev\sdv-mods\ImmersiveValley\Common\Extensions\Reflection\StringExtensions.cs:line 15
   at DaLion.Stardew.Professions.Framework.Patches.Integrations.UiInfoSuite.ExperienceBarGetExperienceRequiredToLevelPatch..ctor() in C:\dev\sdv-mods\ImmersiveValley\ImmersiveProfessions\Framework\Patches\Integrations\UiInfoSuite\ExperienceBarGetExperienceRequiredToLevelPatch.cs:line 17
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at DaLion.Common.Harmony.Harmonizer.ApplyAll() in C:\dev\sdv-mods\ImmersiveValley\Common\Harmony\Harmonizer.cs:line 67

**Comment by Katsuane** [comment]
Welp now two mods are broken. I love updates for no good reason, thanks guys. Who knows when it'll get fixed now.

**Comment by lightsourced** [comment]
Uhm... Walk of Life author fixed the issue... just update the mod. The updates are good, they introduce performance optimizations or fix bugs. No need to get nasty.

**Comment by M-DaLion** [comment]
Sorry Annosz.
I've since added some context to the error message so people know not to report here (at least those who read).

**Comment by lightsourced** [comment]
Yea, sorry, posted this before I saw your message on the mod page :)

**Comment by Annosz** [comment]
Hey @M-DaLion, thanks for popping in! :) It's not a problem, even I didn't know which mod has a problem when there is a compatibility problem. Also @lightsourced it was perfectly okay to write here too :)

I am planning to do more renaming in the future too, as I saw this error was caused by changing the `UIInfoSuite.UIElements.ExperienceBar` namespace, right? I will try to let you know @M-DaLion when any other change takes place with that class.

Also, you are really cool, sadly I only met with your mod yesterday, and it's incredible work you do there with the support! :) I envy your dedication, wish I was putting out versions as fast as you!
